include/nbl/builtin/hlsl/scan/declarations.hlsl

```diff
@@ -0,0 +1,66 @@
+#ifndef _NBL_HLSL_SCAN_DECLARATIONS_INCLUDED_
+#define _NBL_HLSL_SCAN_DECLARATIONS_INCLUDED_
+
+// REVIEW: Not sure if this file is needed in HLSL implementation
+
+#include "nbl/builtin/hlsl/scan/parameters_struct.hlsl"
+
+
+#ifndef _NBL_HLSL_SCAN_GET_PARAMETERS_DECLARED_
+namespace nbl
+{
+namespace hlsl
+{
+namespace scan
+{
+	Parameters_t getParameters();
```

Comment:
no they are not needed, since now you can have 3 functors as struct types as templates on your scanning functions
```
template<typename ParameterAccessor, typename DataGetter, typename DataSetter>
```